scripts/data/unified_tokenize_dataset.py

```diff
@@ -212,6 +219,24 @@ def _process_verifiable_answer_sample(self, sample: Any):
             'verified_answer': verified_answer,
         }
 
+    def _process_verifiable_answer_with_metadata_sample(self, sample: Any):
+        """Process a prompt sample with metadataand extract the answer.
+
+        Args:
+            sample (Any): a sample from the dataset
+
+        Returns:
+            dict: a dictionary containing the prompt, verified answer, and metadata
+        """
+        metadata = sample.get('metadata', None)
+        if metadata is None:
+            log.warning(f'No metadata found for sample: {sample}')
+            return None
+
+        sample = self._process_verifiable_answer_sample(sample)
+        sample['metadata'] = metadata
+        return sample
+
```

Comment:
not convinced this couldn't just be a part of `_process_verifiable_answer_sample`.